archivy/check_changes.py

```diff
@@ -85,21 +86,29 @@ def on_created(self, event):
                 self.format_file(event.src_path)
 
 
-def run_watcher(app):
-    event_handler = ModifHandler(app)
-    observer = Observer()
-    observer.schedule(
-        event_handler,
-        path=os.path.join(
-            app.config['APP_PATH'],
-            'data' + SEP
-        ),
-        recursive=True)
-    observer.start()
-
-    try:
-        while True:
+class Watcher(Thread):
+
+    def __init__(self, app):
+        self.running = True
+        self.app = app
+        Thread.__init__(self)
+
+    def run(self):
+        event_handler = ModifHandler(self.app)
+        observer = Observer()
+        observer.schedule(
+            event_handler,
+            path=os.path.join(
+                self.app.config['APP_PATH'],
+                'data' + SEP
+            ),
+            recursive=True)
+        observer.start()
+
+        while self.running:
             time.sleep(1)
-    except KeyboardInterrupt:
         observer.stop()
-    observer.join()
+        observer.join()
```

Comment:
I guess `observer.join` should be called before `observer.stop`

Reply:
I'm taking from the lib's example: https://python-watchdog.readthedocs.io/en/v0.10.2/quickstart.html#a-simple-example

Reply:
Yes, but that example has `.join` inside the `try:` block. 
So the minimum or basic code flow is 
`t.join() -> t.stop() -> t.join()`

But in this case, its this
`t.stop() -> t.join() `

Reply:
Ok, nvm. I am wrong. You are right :sweat_smile:

Reply:
I wonder where the `Mark as resolved` button went. anyway, its resolved :rocket:

Reply:
So should it become `t.join()` -> `t.stop()`?